sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py

```diff
@@ -443,14 +500,22 @@ def generate_share_sas(
             raise ValueError("'expiry' parameter must be provided when not using a stored access policy.")
         if not permission:
             raise ValueError("'permission' parameter must be provided when not using a stored access policy.")
-    sas = FileSharedAccessSignature(account_name, account_key)
+    if not user_delegation_key and not account_key:
+        raise ValueError("Either user_delegation_key or account_key must be provided.")
+    if isinstance(account_key, UserDelegationKey):
+        user_delegation_key = account_key
+    if user_delegation_key:
+        sas = FileSharedAccessSignature(account_name, user_delegation_key=user_delegation_key)
+    else:
+        sas = FileSharedAccessSignature(account_name, account_key=account_key)
```

Comment:
Also copied from Blob but this if-statement also doesn't seem necessary... Just pass both `account_key` and `user_delegation_key`, one should be `None`. You can probably make the `user_delegation_key` param in the `FileSharedAccessSignature` ctor non-optional in that case.